i18n/babel-plugin.js

```diff
@@ -76,27 +76,54 @@ const DEFAULT_OUTPUT = 'gettext.pot';
 const VALID_TRANSLATION_KEYS = [ 'msgid', 'msgid_plural', 'msgctxt' ];
 
 /**
- * Returns translator comment for a given AST node if one exists.
+ * Regular expression matching translator comment value.
  *
- * @param  {Object}  node AST node
- * @return {?string}      Translator comment
+ * @type {RegExp}
  */
-function getTranslatorComment( node ) {
-	if ( ! node.leadingComments ) {
-		return;
+const REGEXP_TRANSLATOR_COMMENT = /^\s*translators:\s*([\s\S]+)/im;
+
+/**
+ * Returns translator comment for a given AST traversal path if one exists.
+ *
+ * @param  {Object}  path              Traversal path
+ * @param  {Number}  _originalNodeLine Private: In recursion, line number of
+ *                                     the original node passed
+ * @return {?string}                   Translator comment
+ */
+function getTranslatorComment( path, _originalNodeLine ) {
+	const { node, parent, parentPath } = path;
+
+	// Assign original node line so we can keep track in recursion whether a
+	// matched comment or parent occurs on the same or previous line
+	if ( ! _originalNodeLine ) {
+		_originalNodeLine = node.loc.start.line;
 	}
 
-	const comments = node.leadingComments.reduce( ( memo, comment ) => {
-		const match = comment.value.match( /^\s*translators:\s*(.*?)\s*$/im );
-		if ( match ) {
-			memo.push( match[ 1 ] );
-		}
+	const comment = find( node.leadingComments, ( commentNode ) => {
+		const { line } = commentNode.loc.end;
+		return (
+			line >= _originalNodeLine - 1 &&
+			line <= _originalNodeLine &&
+			REGEXP_TRANSLATOR_COMMENT.test( commentNode.value )
```

Comment:
An earlier iteration did this, but alas I encountered that `find` returns the entry (i.e. entire node) for which the callback returns true, not the return value itself. I might poke around for alternatives though, or if you have any suggestions.